scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -419,9 +419,22 @@ abstract class FixedPathSource(path: String*) extends FileSource {
   def localPaths = path.toList
   def hdfsPaths = path.toList
 
+  override def hdfsWritePath = stripTrailingStar(super.hdfsWritePath)
+
   override def toString = getClass.getName + path
   override def hashCode = toString.hashCode
   override def equals(that: Any): Boolean = (that != null) && (that.toString == toString)
+
+  /**
+    * Strip trailing '*' from the path string.
+    */
+  protected def stripTrailingStar(path: String): String = {
+    if(path.takeRight(2) == "/*") {
```

Comment:
Yeah I was wondering the same as well. Getting rid of the '*' would seems like it wouldn't work. We could potentially replace the "*" with "." or error out like @johnynek suggested.